### PR TITLE
[ci] Add Dockerfile and Kokoro configuration for Bazel and SwiftShader

### PR DESCRIPTION
**Pull request by antiagainst**
This commit adds Dockerfile and Kokoro configuration to run
IREE's TensorFlow integration tests on SwiftShader.

### CODE REVIEW

**Comment by antiagainst** [comment]
Again, it might be good to bake SwiftShader into the base bazel image and have it always available and cover GPU tests. But that can happen later.

File: build_tools/kokoro/gcp_ubuntu/bazel/linux/x86/swiftshader/continuous.cfg [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Format: //devtools/kokoro/config/proto/build.proto
```
Comment (GMNGeoffrey) [inline build_tools/kokoro/gcp_ubuntu/bazel/linux/x86/swiftshader/continuous.cfg:1]: This file isn't necessary. There is no "continuous" job anymore (it's been replaced by the "main" and "google" jobs). Looks like I forgot to clean up the leftover files. Opened #2649

File: build_tools/kokoro/gcp_ubuntu/bazel/linux/x86/swiftshader/build_kokoro.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
```
Comment (GMNGeoffrey) [inline build_tools/kokoro/gcp_ubuntu/bazel/linux/x86/swiftshader/build_kokoro.sh:1]: Given current conventions, this should be: `build_tools/kokoro/gcp_ubuntu/bazel/linux/x86-swiftshader/integrations/build_kokoro.sh` See also my comments on #2546

File: build_tools/docker/bazel_swiftshader/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+# Run interactively using the following, where IREE_WORKDIR is the path to your
+# local dev environment:
+# docker run -it --rm --entrypoint bash --volume ${IREE_WORKDIR}:/usr/src/iree/ \
+#   gcr.io/iree-oss/bazel-swiftshader
+
+# Set up the image and working directory.
+FROM gcr.io/iree-oss/bazel-tensorflow
+
+# TODO: The following steps are copied from cmake, cmake-vulkan, and
```
Comment (GMNGeoffrey) [inline build_tools/docker/bazel_swiftshader/Dockerfile:30]: Can you open a tracking issue?

File: build_tools/docker/bazel_swiftshader/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+ARG SWIFTSHADER_COMMIT=6a8a74986c357b0c6fa0dfd2b4b9230af8d39d1a
+
+# Then compile and install SwiftShader.
+
+# cmake, ninja, and zlib is needed for compiling SwiftShader.
+RUN apt-get update && apt-get install -y cmake ninja-build zlib1g-dev
+
+# Update cmake to v3.13+, which is ahead of apt-get's version (3.10.2).
+# Install dependencies, including an old version of cmake to bootstrap.
```
Comment (GMNGeoffrey) [inline build_tools/docker/bazel_swiftshader/Dockerfile:59]: This comment is no longer true

File: build_tools/kokoro/gcp_ubuntu/bazel/linux/x86/swiftshader/continuous.cfg (hunk 1 above)
Comment (antiagainst) [reply build_tools/kokoro/gcp_ubuntu/bazel/linux/x86/swiftshader/continuous.cfg:1]: Yeah. I'm basically just copy paste. :)

File: build_tools/docker/bazel_swiftshader/Dockerfile (hunk 3 above)
Comment (antiagainst) [reply build_tools/docker/bazel_swiftshader/Dockerfile:30]: Done. #2651

File: build_tools/docker/bazel_swiftshader/Dockerfile (hunk 4 above)
Comment (antiagainst) [reply build_tools/docker/bazel_swiftshader/Dockerfile:59]: Removed. BTW, I see kitware has an apt repo: https://apt.kitware.com/. Might be able to get pre-built cmake binaries there directly? I haven't looked into whether they allow you to download a specific version yet though.

File: build_tools/kokoro/gcp_ubuntu/bazel/linux/x86/swiftshader/build_kokoro.sh (hunk 2 above)
Comment (antiagainst) [reply build_tools/kokoro/gcp_ubuntu/bazel/linux/x86/swiftshader/build_kokoro.sh]: Okay. Moved to that path.

File: build_tools/docker/bazel_swiftshader/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+ARG SWIFTSHADER_COMMIT=6a8a74986c357b0c6fa0dfd2b4b9230af8d39d1a
+
+# Then compile and install SwiftShader.
+
+# cmake, ninja, and zlib is needed for compiling SwiftShader.
+RUN apt-get update && apt-get install -y cmake ninja-build zlib1g-dev
+
+# Update cmake to v3.13+, which is ahead of apt-get's version (3.10.2).
+ENV CMAKE_VERSION 3.13.5
+RUN apt-get update \
```
Comment (phoenix-meadowlark) [inline build_tools/docker/bazel_swiftshader/Dockerfile:60]: Could we do this in our `Dockerfile`s with the following?

```shell
snap install cmake --classic
```

That installs `cmake 3.18.0`. I don't know much about `snap`, just saw Thomas use it in a doc.

Comment (antiagainst) [reply build_tools/docker/bazel_swiftshader/Dockerfile:60]: I think Kitware has an apt repo: https://apt.kitware.com/. We might be able to get specific version of CMake from it. But I haven't tried it yet.

Anyway, this sounds like something we can clean up later together. I just copied it from existing Dockerfile. :)

Comment (phoenix-meadowlark) [reply build_tools/docker/bazel_swiftshader/Dockerfile:60]: Sounds good!

**Comment by phoenix-meadowlark** [comment]
Not controlled by this PR, but should `linux_x86-swiftshader_bazel` be named `linux_x86-swiftshader_bazel_integrations` if it's building the integrations?

**Comment by GMNGeoffrey** [comment]
> Not controlled by this PR, but should `linux_x86-swiftshader_bazel` be named `linux_x86-swiftshader_bazel_integrations` if it's building the integrations?

Yes it should. I'll fix that. Actually they should all be renamed to match the directory structure, but I'm waiting till all the PRs branched off before that restructure are in to redo the job configs and badge names and such.

**Pull request merged**